ctru-rs/src/applets/mii_selector.rs

```diff
@@ -0,0 +1,193 @@
+use crate::mii::MiiData;
+use bitflags::bitflags;
+use std::ffi::CString;
+
+#[derive(Debug, Clone)]
+pub enum MiiConfigIndex {
+    Index(u32),
+    All,
+}
+
+#[derive(Debug, Clone)]
+pub enum MiiType {
+    Guest { index: u32, name: String },
+    User,
+}
+
+bitflags! {
+    pub struct Options: u32 {
+        const MII_SELECTOR_CANCEL = ctru_sys::MIISELECTOR_CANCEL;
+        const MII_SELECTOR_GUESTS = ctru_sys::MIISELECTOR_GUESTS;
+        const MII_SELECTOR_TOP = ctru_sys::MIISELECTOR_TOP;
+        const MII_SELECTOR_GUEST_START = ctru_sys::MIISELECTOR_GUESTSTART;
+    }
+}
+
+pub struct MiiSelector {
+    config: Box<ctru_sys::MiiSelectorConf>,
+}
+
+pub struct MiiSelectorReturn {
+    raw_return: Box<ctru_sys::MiiSelectorReturn>,
+    pub mii_data: MiiData,
+    pub is_mii_selected: bool,
+    pub mii_type: MiiType,
+    pub checksum: u16,
+}
+
+impl MiiSelector {
+    pub fn init() -> Self {
+        let mut config = Box::<ctru_sys::MiiSelectorConf>::default();
+        unsafe {
+            ctru_sys::miiSelectorInit(config.as_mut());
+        }
+        Self { config }
+    }
+
+    pub fn set_title(&mut self, text: &str) {
+        // This can only fail if the text contains NUL bytes in the string... which seems
+        // unlikely and is documented
+        let c_text = CString::new(text).expect("Cstring::new failed");
+        unsafe {
+            ctru_sys::miiSelectorSetTitle(self.config.as_mut(), c_text.as_ptr());
+        }
+    }
+
+    pub fn set_options(&mut self, options: Options) {
+        unsafe { ctru_sys::miiSelectorSetOptions(self.config.as_mut(), options.bits) }
+    }
+
+    pub fn whitelist_guest_mii(&mut self, mii_index: MiiConfigIndex) {
+        match mii_index {
+            MiiConfigIndex::Index(i) => unsafe {
+                ctru_sys::miiSelectorWhitelistGuestMii(self.config.as_mut(), i)
+            },
+            MiiConfigIndex::All => unsafe {
+                ctru_sys::miiSelectorWhitelistGuestMii(
+                    self.config.as_mut(),
+                    ctru_sys::MIISELECTOR_GUESTMII_SLOTS,
+                )
+            },
+        }
```

Comment:
nit: Could reduce duplication (this applies to multiple methods):
```suggestion
        let index = match mii_index {
            MiiConfigIndex::Index(i) => i,
            MiiConfigIndex::All => ctru_sys::MIISELECTOR_GUESTMII_SLOTS,
        };
        unsafe { ctru_sys::miiSelectorWhitelistGuestMii(self.config.as_mut(), index) };
```

Even better, move the `match` conversion to a method on `MiiConfigIndex`.

Reply:
> Even better, move the match conversion to a method on MiiConfigIndex.

The problem with that is that, depending on the context, ``MiiConfigIndex::All`` returns ``MIISELECTOR_GUESTMII_SLOTS`` or ``MIISELECTOR_USERMII_SLOTS``

Reply:
In that case just doing the refactor in each method would be good.